File: AppEmag/Models/ServerError/ServerError.m
//
//  ServerError.m
//  AppEmag
//
//  Created by Viorel on 07.09.2023.
//

#import <Foundation/Foundation.h>
#import "ServerError.h"

@implementation ServerError

- (instancetype)initWithDictionary:(NSDictionary *)dictionary {
    self = [super init];
    if (self) {
        _errorCode = [dictionary objectForKey:@"errorCode"];
        _errorMessage = [dictionary objectForKey:@"errorMessage"];
    }
    return self;
}

- (void)encodeWithCoder:(NSCoder *)encoder {
    [encoder encodeObject:self.errorCode forKey:@"errorCode"];
    [encoder encodeObject:self.errorMessage forKey:@"errorMessage"];
}

- (instancetype)initWithCoder:(NSCoder *)decoder {
    self = [super init];
    if (self) {
        _errorCode = [decoder decodeObjectForKey:@"errorCode"];
        _errorMessage = [decoder decodeObjectForKey:@"errorMessage"];
    }
    return self;
}

@end

File: AppEmag/Models/ServerError/ServerError.swift
//
//  ServerError.swift
//  AppEmag
//
//  Created by Viorel on 06.09.2023.
//

import Foundation

struct ServerError: Codable {
    let errorCode: String
    let errorMessage: String

    enum CodingKeys: String, CodingKey {
        case errorCode = "error_code"
        case errorMessage = "error_message"
    }
}
